packages/@aws-cdk/aws-iam/lib/oidc-provider.ts

```diff
@@ -0,0 +1,154 @@
+import { Construct, CustomResource, CustomResourceProvider, CustomResourceProviderRuntime, IResource, Resource, Stack, Token } from '@aws-cdk/core';
+import * as path from 'path';
+
+const RESOURCE_TYPE = 'Custom::AWSCDKOpenIdConnectProvider';
+
+/**
+ * Represents an IAM OpenID Connect provider.
+ *
+ * @experimental
+ */
+export interface IOpenIdConnectProvider extends IResource {
+  /**
+   * The Amazon Resource Name (ARN) of the IAM OpenID Connect provider.
+   */
+  readonly openIdConnectProviderArn: string;
+}
+
+/**
+ * Initialization properties for `OpenIdConnectProvider`.
+ * @experimental
+ */
+export interface OpenIdConnectProviderProps {
+  /**
+   * The URL of the identity provider. The URL must begin with https:// and
+   * should correspond to the iss claim in the provider's OpenID Connect ID
+   * tokens. Per the OIDC standard, path components are allowed but query
+   * parameters are not. Typically the URL consists of only a hostname, like
+   * https://server.example.org or https://example.com.
+   *
+   * You cannot register the same provider multiple times in a single AWS
+   * account. If you try to submit a URL that has already been used for an
+   * OpenID Connect provider in the AWS account, you will get an error.
+   */
+  readonly url: string;
+
+  /**
+   * A list of client IDs (also known as audiences). When a mobile or web app
+   * registers with an OpenID Connect provider, they establish a value that
+   * identifies the application. (This is the value that's sent as the client_id
+   * parameter on OAuth requests.)
+   *
+   * You can register multiple client IDs with the same provider. For example,
+   * you might have multiple applications that use the same OIDC provider. You
+   * cannot register more than 100 client IDs with a single IAM OIDC provider.
+   *
+   * Client IDs are up to 255 characters long.
+   *
+   * @default - no clients are allowed
+   */
+  readonly clientIds?: string[];
+
+  /**
+   * A list of server certificate thumbprints for the OpenID Connect (OIDC)
+   * identity provider's server certificates.
+   *
+   * Typically this list includes only one entry. However, IAM lets you have up
+   * to five thumbprints for an OIDC provider. This lets you maintain multiple
+   * thumbprints if the identity provider is rotating certificates.
+   *
+   * The server certificate thumbprint is the hex-encoded SHA-1 hash value of
```

Comment:
Uppercase or lowercase, if that matters?

Reply:
I don't know. This is basically a copy & pasta from AWS docs.